### ISSUE TITLE
DualShock 4 (PS4) controller not working

### ISSUE THREAD

**Issue by Lastique** [ValveSoftware/Proton#5502]
# Compatibility Report
- Name of the game with compatibility issues: Presumably, any Windows game running under Proton. Tried Horizon Zero Dawn, Mass Effect Legendary Edition, Thief (2014)
- Steam AppID of the game: 1151640, 1328670, 239160

## System Information

- Kubuntu 21.10
- GPU: GeForce RTX 2080 Ti
- Driver/LLVM version: Nvidia 510.39.01
- Kernel version: 5.13.0-25-lowlatency
- Link to full system information report as [Gist](https://gist.github.com/): [steam_system_info.txt](https://github.com/ValveSoftware/Proton/files/7885021/steam_system_info.txt)
- Proton version: 6.3-8

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

## Symptoms

Upon launching the game, the controller LED lights up blue (normally it is turned off on my system after connection). In the game, buttons or stick movements are not recognized. The touchpad area acts as a mouse touchpad.

The gamepad is working correctly in native games running from Steam, as well as Windows games running under vanilla Wine 7.0-rc6.

## Reproduction

1. Connect DualShock 4 over Bluetooth.
2. Launch a game via Proton.
3. Try interacting with the game using the controller (e.g. navigate the in-game menu).

**Comment by Lastique** [comment]
The controller LED lighting up blue is possibly a Steam client issue, since it also happens when the client (not the game) starts up. See https://github.com/ValveSoftware/steam-for-linux/issues/8019.

**Comment by ipr** [comment]
In some cases you need to exit game and launch it again, notably after Proton version has changed (such as 6.3 -> 7.0).

**Comment by Lastique** [comment]
I had been launching the game multiple times, the controller wasn't working every time. I haven't tested with Proton 7.0 though.

**Comment by ipr** [comment]
> I had been launching the game multiple times, the controller wasn't working every time. I haven't tested with Proton 7.0 though.

I know there are games where it just doesn't work for some reason, but I was making a note that something in the launch of a newer version might need improvement. Some games with Uplay have issues where you need to disable Steam input to get gamepad working so there are some rough edges somewhere.

Thief (2014) works fine with Proton 7.0 for me.

**Comment by Lastique** [comment]
This problem still persists. With Proton 7.0 too.

**Comment by skdza** [comment]
Any updates or fixes?

**Comment by Arroquw** [comment]
Having this issue as well, it works wired but not over bluetooth.

**Comment by simifor** [comment]
my controller lits up blue and it worked fine for a couple of titles I tried: brotato, ys viii and deep rock galactic. Is the controller recognized by the system when connected through bluetooth? Something like `antimicrox` can be used to confirm the controller itself and its input is being recognized by the system.

Another factor could be bluez, I tried with 5.79 and git and both worked fine.

**Comment by Lastique** [comment]
> Is the controller recognized by the system when connected through bluetooth?

As I wrote in the bug description, the controller works other than in Proton games.

**Comment by simifor** [comment]
@Lastique yes, but recent reports didn't specify, so it can be hard to determine what is currently affecting users. As far as the led goes, mine remains blue while it's connected to my computer, with or without a game being open. Tested thief 4 just now and it worked as expected.

What do your steam input settings look like? Is playstation support enabled?
The only other thing I could think of possibly interfering is some udev rule, they are usually at `/usr/lib/udev/rules.d/`

**Comment by Lastique** [comment]
> What do your steam input settings look like? Is playstation support enabled?

I don't have access to that system at the moment, but AFAIR I didn't enable any Steam Input functionality. That is, Steam Input should be disabled and the controller input should be interpreted as is, without any transformation by Steam.

BTW, the blue LED light seems to be a generic Wine+Linux kernel issue as I see the blue LED light up even when running apps under Wine, and it started happening after an upgrade to some Ubuntu version. Looks like Wine initializes the controller on startup and at some point the kernel started initializing the LED on that event.

**Comment by matheustavarestrindade** [comment]
I'm having the same issue here, on borderlands 4 it dont work, also on The last of us 2 remake.
